### PR TITLE
Audit and Fixed text-files.md

### LINKED ISSUE #122: Audit and fix E-Secondary-Storage/text-files.md
- Look for any typos
- Make sure all Markdown is correctly done.  The conversion was done automatically by a tool, so there may be more idiomatic ways to do some things
- Check the page in a Desktop browser.  Does it look OK?  Are there any issues that need to be fixed?
- Check the page in a Mobile browser.  Does it look OK?  Are there any issues that need to be fixed?
- Try the page in both Light and Dark mode.  Are there any issues that need to be fixed?
- Try running the page through Lighthouse: https://developers.google.com/web/tools/lighthouse.  Are there any issues that need to be fixed?
- Try running the page through Web Hint: https://webhint.io/.  Are there any issues that need to be fixed?
- Tables that are really wide (types.md) could be converted to use SVG or flex/grid. Make wide visual tables fit in the available viewport.  Another example is the table in `docs/D-Modularity/pointers.md`, where we should reduce the vertical padding/margins on tables. The tables are too large, and probably shouldn't be tables at all.
- Prefer Markdown tables vs. raw HTML tags
- Fix colours in tables so it uses global styles vs. inline styles
- Use alt tags for all images.  See https://supercooldesign.co.uk/blog/how-to-write-good-alt-text for suggestions
- Avoid blockquote for indentation, and prefer q or blockquote only when actually quoting.  Use Admonitions instead. Example: `docs/D-Modulatirty/pointers.md`: there are places where > Blockquote are being used and Admonitions could replace it. “Note:” Lightbulb info Admonition.
- Make sure images work in dark and light modes.  For example, `docs/8-Computations/logic.md` Flags image is white background, needs to work in dark.  Create a “negative” or SVG so we can do Dark Mode?  Can we programmatically create this negative?
- Change colours for tables (don’t use Yellow).  We should use the default colour theme.
- Avoid using `<b>` and `<i>` to do emphasis.
- Use backticks and all code for memory addresses and such in tables that use things like void, int, function names, memory addresses.
- Make sure all code blocks use syntax highlighting
- centre images horizontally (esp. when they are small) so they work well at all responsive sizes
- check sub-script and super-script that they are correctly used in all cases
- consider including inter-site links to other pages that have related info
- all links should have descriptive text for accessibility
- Make sure that sub-headings (right-hand side) for each document are correct.
- Improve readability of code comments (i.e., `//` and `/*...*/`) in dark mode (even light mode might need some work)
- In places where `***...***` was used as a fourth heading level, use `####` instead.
- Fix Frontmatter for page to include proper id, title, slug, etc.

### PR DESCRIPTION
**Pull request by joshuali7536**
Fixes #122

- Fixed typos and grammar
- Checked Desktop, Mobile, Light & Dark
- Checked Lighthouse and Web Hint
- Added better alt tags for images
- Changed blockquotes to use Admonitions
- Added inter-site links
- Added Frontmatter

### CODE REVIEW

**Comment by Kevan-Y** [comment]
@joshuali7536 Hi, there is a recent change where we remove `slug` (#142).  Could you kindly remove slug in all you changed files, thank you

**Comment by joshuali7536** [comment]
@Kevan-Y the `slug` has been removed from the modified files.

File: docs/E-Secondary-Storage/text-files.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
 ---
+id: text-files
+title: Text Files
 sidebar_position: 1
 ---
```
Comment (Kevan-Y) [inline docs/E-Secondary-Storage/text-files.md:5]: Add a description here, check the standard https://github.com/Seneca-ICTOER/IPC144/pull/143

**Review by Kevan-Y — CHANGES_REQUESTED** [review]

**Comment by humphd** [comment]
This is failing CI (yeah @mqnguyen5!):

```
 Unable to build website for locale "en".
Error: Docusaurus found broken links!

Please check the pages of your site in the list below, and make sure you don't reference any path that does not exist.
Note: it's possible to ignore broken links with the 'onBrokenLinks' Docusaurus configuration, and let the build pass.

Exhaustive list of all broken links found:

- On source page path = /E-Secondary-Storage/text-files:
   -> linking to ../introduction/information (resolved as: /introduction/information)
   -> linking to ../modularity/pointers (resolved as: /modularity/pointers)
   -> linking to ../data-structures/structures (resolved as: /data-structures/structures)
```

**Comment by humphd** [comment]
@CameronGray1210 this is a good example of how our new CI stuff works, in this case, catching a broken link introduced in a PR: https://github.com/Seneca-ICTOER/IPC144/runs/4459967013?check_suite_focus=true.

**Comment by joshuali7536** [comment]
Links have been modified to the correct links.

**Comment by Kevan-Y** [comment]
Could you rebase this branch to the latest main commit?

**Review by CameronGray1210 — APPROVED** [review]
Good job - thanks!

**Pull request merged**